File: src/main/java/com/manifestors/shinrai/client/command/commands/BindCommand.java
package com.manifestors.shinrai.client.command.commands;

import com.manifestors.shinrai.client.Shinrai;
import com.manifestors.shinrai.client.command.Command;
import com.manifestors.shinrai.client.utils.input.GLFWUtil;

public class BindCommand extends Command {

    public BindCommand() {
        super("bind", "Change our reset modules keybind.", "moduleName, key", "b");
    }

    @Override
    public boolean onCommandExecuted(String[] args) {
        if (args.length < 3)
            return false;

        var moduleName = args[1];
        var keyInput = args[2];

        var module = Shinrai.INSTANCE.getModuleManager().getModuleByName(moduleName);

        if (module != null) {
            if (keyInput.equalsIgnoreCase("none")) {
                module.setKeyCode(0);
                Shinrai.INSTANCE.addChatMessage("Unbound module " + module.getName() + ".");
            } else {
                var keyName = keyInput.toUpperCase();
                var key = GLFWUtil.getKeyIndex(keyName);
                if (key != -1) {
                    module.setKeyCode(key);
                    Shinrai.INSTANCE.addChatMessage(module.getName() + " bound to " + keyName + ".");
                } else
                    Shinrai.INSTANCE.addChatMessage("Invalid key " + keyInput + ".");
            }
        } else {
            sendNotFound("Module '" + moduleName + "'");
        }

        return true;
    }
}

File: src/main/kotlin/com/manifestors/shinrai/client/command/commands/BindCommand.kt
package com.manifestors.shinrai.client.command.commands

import com.manifestors.shinrai.client.Shinrai.addChatMessage
import com.manifestors.shinrai.client.Shinrai.moduleManager
import com.manifestors.shinrai.client.command.Command
import com.manifestors.shinrai.client.utils.input.GLFWUtil.getKeyIndex
import java.util.*

class BindCommand : Command(
    "bind",
    "Change our reset modules keybind.",
    "moduleName, key",
    "b") {
    override fun onCommandExecuted(args: Array<String>): Boolean {
        if (args.size < 3) return false

        val (moduleName, keyInput) = args.drop(1)

        moduleManager.getModuleByName(moduleName)?.let {
            if (keyInput.equals("none", ignoreCase = true)) {
                it.keyCode = 0
                addChatMessage("Unbound module " + it.name + ".")
            } else {
                val keyName = keyInput.uppercase(Locale.getDefault())
                val key = getKeyIndex(keyName)
                if (key != -1) {
                    it.keyCode = key
                    addChatMessage(it.name + " bound to " + keyName + ".")
                } else addChatMessage("Invalid key '$keyInput'.")
            }
        } ?: sendNotFound("Module '$moduleName'")

        return true
    }
}
